wallet/src/de/schildbach/wallet/ui/DashPayUserActivity.kt

```diff
@@ -63,13 +77,24 @@ class DashPayUserActivity : InteractionAwareActivity() {
             intent.putExtra(CONTACT_REQUEST_RECEIVED, contactRequestReceived)
             return intent
         }
+
+        @JvmStatic
+        fun createIntent(context: Context, userId: String): Intent {
+            val usernameSearchResult = runBlocking { PlatformRepo.getInstance().getLocalUsernameSearchResult(userId) }
```

Comment:
getLocalUsernameSearchResult actually makes three queries to the database.  A MediatorLiveData, i think, could be used instead.

Reply:
though the calling method should perhaps call the access livedata.